src/main/java/com/scalar/cassy/transferer/AzureBlobFileDownloader.java

```diff
@@ -32,39 +30,29 @@ public void download(RestoreConfig config) {
     String key = BackupPath.create(config, config.getKeyspace());
 
     logger.info("Downloading " + blobContainerClient.getBlobContainerName() + "/" + key);
-    List<Mono<BlobProperties>> filesToBeDownloaded = new ArrayList<>();
-    Iterable<BlobItem> keyspaceBlobs =
-        blobContainerClient.listBlobs(new ListBlobsOptions().setPrefix(key + "/")).toIterable();
-    for (BlobItem blob : keyspaceBlobs) {
+    Iterator<BlobItem> keyspaceBlobs =
+            blobContainerClient.listBlobs(new ListBlobsOptions().setPrefix(key + "/"), null).iterator();
+    while (keyspaceBlobs.hasNext()) {
+      BlobItem blob = keyspaceBlobs.next();
       Path destFile = Paths.get(config.getDataDir(), blob.getName());
+
       try {
         Files.createDirectories(destFile.getParent());
       } catch (IOException e) {
         throw new FileTransferException(e);
       }
-      filesToBeDownloaded.add(
-          blobContainerClient
-              .getBlobAsyncClient(blob.getName())
-              .downloadToFile(destFile.toString())
-              .doOnSuccess(
-                  blobProperties -> logger.info("Download file succeeded : " + destFile.toString()))
-              .doOnError(
-                  error -> {
-                    throw new FileTransferException(
-                        "Download file failed : " + destFile.toString(), error);
-                  }));
 
-      if (filesToBeDownloaded.size() >= ASYNC_FILE_DOWNLOAD_LIMIT) {
-        // Start downloading files asynchronously and wait for them to complete
-        Mono.when(filesToBeDownloaded).block();
-        filesToBeDownloaded.clear();
+      try {
+        try (OutputStream outputStream = new FileOutputStream(destFile.toString())) {
+          blobContainerClient
```

Comment:
Can we check async works?
Synchronous downloading one by one takes too much time, right?

Reply:
> Can we check async works?

Ok, I will check

> Synchronous downloading one by one takes too much time, right?

Yes,

80GB download took 21 min.
64GB download took 15 min